src/traces/scatterregl/plot.js

```diff
@@ -0,0 +1,575 @@
+    this.range = this.scatterOptions.range = this.lineOptions.range = range;
+
+    this.regl._refresh();
+    this.scatter({
+        range: range
+    });
+    this.regl._refresh();
+    this.line({
+        range: range
+    });
+}
+
+
+proto.update = function(cdscatter) {
+    var options = cdscatter[0].trace;
+
+    var container = this.container,
+        xaxis = Axes.getFromId(container, options.xaxis || 'x'),
+        yaxis = Axes.getFromId(container, options.yaxis || 'y'),
+        bounds = this.bounds,
+        selection = options.selection;
+    var vpSize = container._fullLayout._size,
+        width = container._fullLayout.width,
+        height = container._fullLayout.height;
+
+    // makeCalcdata runs d2c (data-to-coordinate) on every point
+    var x = this.pickXData = xaxis.makeCalcdata(options, 'x').slice();
+    var y = this.pickYData = yaxis.makeCalcdata(options, 'y').slice();
+
+    this.xData = x.slice();
+    this.yData = y.slice();
+
+    if(options.visible !== true) {
+        this.isVisible = false;
+        this.hasLines = false;
+        this.hasErrorX = false;
+        this.hasErrorY = false;
+        this.hasMarkers = false;
+    }
+    else {
+        this.isVisible = true;
+        this.hasLines = subTypes.hasLines(options);
+        this.hasErrorX = options.error_x.visible === true;
+        this.hasErrorY = options.error_y.visible === true;
+        this.hasMarkers = subTypes.hasMarkers(options);
+    }
+
+    this.textLabels = options.text;
+    this.name = options.name;
+    this.hoverinfo = options.hoverinfo;
+    this.bounds = [Infinity, Infinity, -Infinity, -Infinity];
+    this.connectgaps = !!options.connectgaps;
+
+    // update viewport & range
+    this.viewport = this.scatterOptions.viewport = this.lineOptions.viewport = [
+        vpSize.l + xaxis.domain[0] * vpSize.w,
+        vpSize.b + yaxis.domain[0] * vpSize.h,
+        (width - vpSize.r) - (1 - xaxis.domain[1]) * vpSize.w,
+        (height - vpSize.t) - (1 - yaxis.domain[1]) * vpSize.h
+    ];
+
+    this.range = this.scatterOptions.range = this.lineOptions.range = [xaxis._rl[0], yaxis._rl[0], xaxis._rl[1], yaxis._rl[1]];
+
+    if(!this.isVisible) {
+        // this.line.clear();
+        // this.errorX.clear();
+        // this.errorY.clear();
+        // this.scatter();
+        // this.fancyScatter.clear();
+    }
+
+    // get error values
+    var errorVals = ErrorBars.calcFromTrace(options, container._fullLayout);
+
+    var len = x.length,
+        idToIndex = new Array(len),
+        positions = Array(2 * len),
+        errorsX = new Float64Array(4 * len),
+        errorsY = new Float64Array(4 * len),
+        pId = 0,
+        ptr = 0,
+        ptrX = 0,
+        ptrY = 0;
+
+    var getX = (xaxis.type === 'log') ? xaxis.d2l : function(x) { return x; };
+    var getY = (yaxis.type === 'log') ? yaxis.d2l : function(y) { return y; };
+
+    var i, xx, yy, ex0, ex1, ey0, ey1;
+
+    for(i = 0; i < len; ++i) {
```

Comment:
I really wish we didn't have to do this here now. Can `regl-scatter2d` and `regl-line2d` just do the right thing will plotly.js `x` and `y` arrays?

Reply:
It can, I just did not finish here :) I will.